### ISSUE TITLE
Update dependency version in envoyproxy/envoy to 0.1.11

### ISSUE THREAD

**Issue by Smeb** [DataDog/dd-trace-cpp#70]
Hi 👋 - hope this is the right place. I opened this issue here rather than in Envoy because it seems like @dgoffredo is a maintainer here, and is the one updating the version in Envoy. If this is not the right place, my apologies.

It seems like Envoy is still using the `0.1.8` version of this library [here](https://github.com/envoyproxy/envoy/blame/180960108843f9867a4ec9a26533df5570cb9c99/bazel/repository_locations.bzl#L538C27-L538C27). This is true as of the latest release (yesterday, 1.19.4). It should probably be updated to `0.1.11`. I'm not too sure how involved this is, or I would open a PR - plus I think it's probably more likely to be accepted by the maintainers here.

**Comment by dgoffredo** [comment]
Hi, Ben. A recent release of dd-trace-cpp contains changes that would break the Datadog-related unit tests within Envoy.

My teammate is working on a PR within Envoy that will make the tests compatible with more recent versions of dd-trace-cpp.

Then, we will update the version of dd-trace-cpp used within Envoy.

Is there a feature or fix in more recent versions of dd-trace-cpp that you're looking for in Envoy?

**Comment by Smeb** [comment]
Great to hear.

The issue we're running into is that with Envoy 1.27 the span names are different than with 1.26. In 1.26 the span names are `envoy.proxy`, while in 1.27 they show up as (what looks to be) the name of the service. This in turn leads some metrics to go missing (specifically trace.envoy.proxy.hits). At least - I think that's what's happening. I haven't got too much hands on experience with tracing / cpp though.

That said, when I dug a bit into the tracing code and old PRs, I saw that the changes [here](https://github.com/envoyproxy/envoy/pull/29932#issuecomment-1772193588) (specifically the change to `setOperation`) don't exist in `release/v1.27` of envoy. I think the backport [here](https://github.com/envoyproxy/envoy/pull/30182/files) was probably missing some things.

It's a little late for me, but would be happy to try updating the backport tomorrow if this sounds like it's the right direction.

**Comment by dgoffredo** [comment]
Yes, the "operation name is wrong in v1.27 and v1.28" was fixed and backported onto those release branches. You can pick up those changes whenever Envoy produces another release in either series. That will fix that problem.

The "setting the 'resource.name' tag doesn't work anymore" issue was fixed on the `main` branch, but the process of proposing backports has [stalled](https://github.com/envoyproxy/envoy/pull/30503). Perhaps I can propose the backport PRs myself.

**Comment by Smeb** [comment]
I haven't worked on Envoy much, but I found [these docs](https://github.com/envoyproxy/envoy/blob/main/RELEASES.md#backports) if it's helpful. Specifically, they say that if you add `backport/review` as a label, that starts the process (apparently).

**Comment by Smeb** [comment]
Just to follow up, I opened backport PRs against 1.27, 1.28, hope that's ok:

- https://github.com/envoyproxy/envoy/pull/30892
- https://github.com/envoyproxy/envoy/pull/30893

Interesting experience getting everything running locally, but tests were passing locally for both patches.

**Comment by njegosrailic** [comment]
Hello everyone,

I work for Wayfair, and we had a support case with DD number `1339031`. I've been following multiple issues in the past two months, and it seems that the problem is finally resolved. If I understood correctly, we would still need new builds for Envoy `1.27.x` and `1.28.x` versions, and they will be included in the next Istio patch versions. Is that correct?

Are you planning to upgrade the [official documentation](https://docs.datadoghq.com/tracing/trace_collection/proxy_setup/?tab=istio#environment-variables) once this is done? It appears that the latest supported version is `v.1.17`, and that was what the support was referring to when we were complaining about traces being _broken_ in Istio `v.1.19`.

Could you also describe any changes related to the DD tracing library that we can expect in Istio versions newer than v1.17, such as the one related to the resource name not being taken from the tag [resource.name](https://github.com/envoyproxy/envoy/pull/29932#issuecomment-1771098114)? These changes may require some of additional adjustments on our side, and we need to communicate this to developers. 
Including these details in the official documentation would help to clarify that it's related to DD Istio integration.

**Comment by dgoffredo** [comment]
Hi, Njegos. I don't know which case number `1339031` refers to. There is Zendesk, which I believe Datadog uses to communicate with customers about Support issues, and then we use another system internally when Support interacts with Engineering. What is the issue that you're referring to with number `1339031`?

The table in the [environment variables][1] section of the documentation you linked is indeed not up to date. I wasn't aware that it exists.

The changes to Envoy that have caused these issues (the unexpected change in operation name, the inability to set the resource name via the "resource.name" tag, and an unintended sampling override) were made to move Envoy from an older OpenTracing-based Datadog tracing library to this newer tracing library (dd-trace-cpp). So, the version numbers tabulated will not be comparable for present and future versions of Istio that use the newer Envoys. On the other hand, dd-trace-cpp has major version zero (0.x.y), so there might not be any confusion.

I'll see to it that the documentation is updated. We have a work item to revisit our Istio documentation more extensively, as methods of deployment have changed since we initially developed the integration.

None of these problematic changes were intentional, and so I can't anticipate what others, if any, might be found. Here's a list of recent Envoy Datadog tracing issues (i.e. issues starting with Envoy v1.27) that I know about:

1. The default Datadog span name changed for spans created by Envoy: https://github.com/envoyproxy/envoy/pull/29932
2. The ability to set the Datadog resource name by setting the "resource.name" tag stopped working: https://github.com/envoyproxy/envoy/pull/30503
3. When Envoy's sampling rate (not to be confused with Datadog's sampling rate) is less then 100%, then drop decisions made by Envoy override any decision extracted from incoming request headers (this can break traces): https://github.com/envoyproxy/envoy/pull/30577

Backports onto Envoy v1.27 and v1.28 are merged for all of these issues ((2) thanks to @Smeb :tada:)

The relationship between Istio and the Datadog tracing library is the following:

![IMG_20231122_153545478](https://github.com/DataDog/dd-trace-cpp/assets/5205367/c04c6700-4856-4094-a34b-da9892097d19)

Istio has a version of Envoy that it calls istio-proxy. istio-proxy is based on a release of Envoy. The correspondence between Istio releases and the Envoy releases on which istio-proxy is based are tabulated [here][2].

Envoy builds dd-trace-cpp from source and combines the library with other [Datadog-specific code][3] within Envoy (this is where I put the bugs).

The version of dd-trace-cpp on which Envoy depends is determined by [envoy/bazel/repository_locations.bzl][4] within the Envoy source tree.

In summary:

- A release of Istio tracks a particular release of Envoy.
- A release of Envoy contains both Datadog-specific code and a particular version of dd-trace-cpp, determined by the build.
- dd-trace-cpp contains the core Datadog tracing code, and is released as a source tarball on GitHub.


[1]: https://docs.datadoghq.com/tracing/trace_collection/proxy_setup/?tab=istio#environment-variables
[2]: https://istio.io/latest/docs/releases/supported-releases/#supported-envoy-versions
[3]: https://github.com/envoyproxy/envoy/tree/main/source/extensions/tracers/datadog
[4]: https://github.com/envoyproxy/envoy/blob/2de016d1007aabff202220b8177167c9ab3e8c6a/bazel/repository_locations.bzl#L553

**Comment by njegosrailic** [comment]
Hi David,

Thank you for the detailed updates and explanations. To be honest, I don't personally use DataDog, but our developers heavily rely on it. In early October, we initiated Istio upgrades, but had to revert in most of the clusters due to the "issue" with tracing. 

Starting from Istio `v1.18`, we observed significant visual differences in Datadog tracing. Some of the main issues include, but are not limited to:

- the function changing from `envoy.proxy` to `<service-name>_<url for cluster>`.
- the `istio.operation` tag seems to be dropped or turned off.
- `runtime-id` and `process_id` are missing from traces.

More details are available in the linked [issue](
https://github.com/envoyproxy/envoy/pull/29932#issuecomment-1824135808). When we approached support, we were informed that this is related to the DataDog tracing library that Envoy uses and that DataDog doesn't support Istio versions beyond v1.17, according to their documentation.

I plan to open a new support case with additional details that I cannot share here. I will mention this issue and then discuss it with our account manager.

Our primary concern is understanding when we can expect the same tracing functionality we had in previous versions. If not, could this be documented?

**Comment by dgoffredo** [comment]
@njegosrailic please share the support case number with me once it is created, so I can inquire about it internally.

Your first bullet point is the "operation name" issue discussed previously. The other two changes of behavior that you mention are news to me. `istio.operation` I haven't heard of before. The `runtime-id` and `process_id` tags are [added][1] by the underlying tracing library, so it is a surprise that they are missing.

[1]: https://github.com/DataDog/dd-trace-cpp/blob/4e490a7b72a8fe5c0f3a49b84db3098c796622ed/src/datadog/trace_segment.cpp#L213-L215

**Comment by njegosrailic** [comment]
Hi @dgoffredo,

I have created a new support case https://help.datadoghq.com/hc/en-us/requests/1453323 and I'm providing the old one https://help.datadoghq.com/hc/en-us/requests/1339031 for reference.

Thank you for the help.

**Comment by dgoffredo** [comment]
This will be resolved when https://github.com/envoyproxy/envoy/pull/31526 is merged.

**Comment by dgoffredo** [comment]
`envoyproxy/envoy` [now uses][1] dd-trace-cpp `v0.1.12`.

[1]: https://github.com/envoyproxy/envoy/pull/31526

**Comment by Smeb** [comment]
Note - this is only true for the latest release (1.29) which isn't currently available in Istio. Should we backport the changes again? The changes look non-trivial, but I guess they're all in the same place.

**Comment by dgoffredo** [comment]
I recommend that we do _not_ backport a newer version of dd-trace-cpp onto Envoy release branches v1.27 and v1.28.

There have been several Datadog tracing problems in Envoy since v1.27, but all of them are due to changes made in Envoy's Datadog extension, not in the separate dd-trace-cpp library.

**Comment by brettcurtis** [comment]
As a customer of Datadog this warning sent me down the path to finding this issue: 

![image](https://github.com/DataDog/dd-trace-cpp/assets/1610100/aa1ae5fb-77b1-4682-9ba6-e958a27fa115)

For me this indicates I should do something about it but after reading this thread I think I'm hearing this is an envoy/DD relationship. If it's true that as a customer I don't have the ability to fix it the warning may be a bit much.

**Comment by brettcurtis** [comment]
Thanks for the details @dgoffredo ! I created an internal support ticket: 1565944 if that helps.
